dbldatagen/schema_parser.py

```diff
@@ -260,6 +260,67 @@ def columnTypeFromString(cls, type_string):
 
         return type_construct
 
+    @classmethod
+    def _cleanseSQL(cls, sql_string):
+        """ Cleanse sql string removing strings so that they are not considered as part of potential column
+            references
+        :param sql_string: String representation of SQL expression
+        :returns: cleansed string
+
+        Any strings identified are replaced with `' '`
+        """
+        assert sql_string is not None, "`sql_string` must be specified"
+
+        # skip over quoted identifiers even if they contain quotes
+        quoted_ident = pp.QuotedString(quoteChar="`", escQuote="``")
+
+        stringForm1 = pp.Literal('r') + pp.QuotedString(quoteChar="'")
+        stringForm2 = pp.Literal('r') + pp.QuotedString(quoteChar='"')
+        stringForm3 = pp.QuotedString(quoteChar="'", escQuote=r"\'")
+        stringForm4 = pp.QuotedString(quoteChar='"', escQuote=r'\"')
+        stringForm = stringForm1 ^ stringForm2 ^ stringForm3 ^ stringForm4
+        stringForm.set_parse_action(lambda s, loc, toks: "' '")
+
+        parser = quoted_ident ^ stringForm
```

Comment:
does quoted_ident also need to have .set_parse_action applied to it for the transform_string to work for that?

Reply:
no it does not. 

What we're looking to do is determine which parts of a sql expression could be field references.

For the expression 'concat('hello', field1, `field 2`)' - field1 and `field 2` could be field references. 

'hello' is not

Reply:
There are some rare cases where a field reference could be inside a literal string - but user can still override with `baseColumn=...`

Goal is to reduce amount of times user has to explicitly declare a reference.